### PR TITLE
Modifications to allow building of toolset in OpenBSD

### PR DESCRIPTION
**Pull request by Ilgrim**
I am using OpenBSD for some microcontroller work and I needed these tools. As far I was not able to compile with the vanilla code, I have modified just a little for be able to compile in OpenBSD 7.4. Build tested in OpenBSD 7.4 and Debian 12.5.

I hope this can be somehow useful.

### CODE REVIEW

**Review by Nightwalker-87 — APPROVED** [review]

File: src/stlink-lib/usb.c [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, enum connect_type connect,
 
         if (ret) { continue; } // could not open device
 
-        uint32_t serial_len = stlink_serial(handle, &desc, sl->serial);
+        uint32_t serial_len = (uint32_t)stlink_serial(handle, &desc, sl->serial);
```
Comment (Ant-ON) [inline src/stlink-lib/usb.c:1175]: I would simply change the type of `serial_len` variable to `size_t`.

File: src/stlink-lib/usb.c [hunk 2]
```diff
@@ -1375,7 +1375,7 @@ static uint32_t stlink_probe_usb_devs(libusb_device **devs, stlink_t **sldevs[],
             break;
         }
 
-        uint32_t serial_len = stlink_serial(handle, &desc, serial);
+        uint32_t serial_len = (uint32_t)stlink_serial(handle, &desc, serial);
```
Comment (Ant-ON) [inline src/stlink-lib/usb.c:1378]: and this too

**Comment by Nightwalker-87** [comment]
@Ilgrim: Please address these points in order to proceed.

**Comment by Ilgrim** [comment]
@Nightwalker-87 Sure! I will make the suggested changes 🙂

**Comment by Nightwalker-87** [comment]
@Ilgrim: I'm looking forward to merge this PR in order to proceed...

File: src/stlink-lib/usb.c (hunk 1 above)
Comment (Nightwalker-87) [reply src/stlink-lib/usb.c:1175]: @Ant-ON: We could alternatively also change stlink_serial() to uint32_t.
This should be safe on all common architectures as well.

**Review by Nightwalker-87 — APPROVED** [review]

**Comment by Nightwalker-87** [comment]
I'll merge this with additional changes afterwards, as missing care on this PR delays unnecessarily delays any further proceedings. :-1:

**Pull request merged**